local_pathfinding/node_mock_global_path.py

```diff
@@ -0,0 +1,220 @@
+                # skip header
+                reader.__next__()
+                for row in reader:
+                    global_path.waypoints.append(
+                        HelperLatLon(latitude=float(row[0]), longitude=float(row[1]))
+                    )
+
+            self.path_mod_tmstmp = path_mod_tmstmp
+            self.file_path = file_path
+
+            # check if global path is just a destination point
+            if len(global_path.waypoints) < 2:
+                interval_spacing = self.get_parameter("interval_spacing")._value
+                pos = self.gps.lat_lon
+
+                msg = MockGlobalPath.generate_path(
+                    dest=global_path.waypoints[0],
+                    interval_spacing=interval_spacing,
+                    pos=pos,
+                    dst_file_path=file_path,
+                )
+            else:
+                msg = global_path
+
+            # publish global path
+            self.global_path_pub.publish(msg)
+            self.get_logger().info(
+                f"Publishing to {self.global_path_pub.topic}: {MockGlobalPath.path_to_dict(msg)}"
+            )
+
+    @staticmethod
+    def generate_path(
+        dest: HelperLatLon,
+        interval_spacing: float,
+        pos: HelperLatLon,
+        dst_file_path: str,
+    ) -> Path:
+        """Returns a path from the current GPS location to the destination point.
+        Waypoints are evenly spaced along the path according to the interval_spacing parameter.
+        Path does not include pos, but does include dest as the final element.
+
+        Args:
+            dest (HelperLatLon): The destination point
+            interval_spacing (float): The desired distance between waypoints on the path
+            pos (HelperLatLon): The current GPS location
+
+        Returns:
+            Path: The generated path
+        """
+        global_path = Path()
+
+        lat1 = pos.latitude
+        lon1 = pos.longitude
+        lat2 = dest.latitude
+        lon2 = dest.longitude
+
+        distance = meters_to_km(GEODESIC.inv(lats1=lat1, lons1=lon1, lats2=lat2, lons2=lon2)[2])
+        n = np.ceil(distance / interval_spacing)
+
+        global_path_tuples = GEODESIC.npts(lon1=lon1, lat1=lat1, lon2=lon2, lat2=lat2, npts=n)
+
+        # npts returns (lon,lat) tuples, its backwards for some reason
+        for lon, lat in global_path_tuples:
+            global_path.waypoints.append(HelperLatLon(latitude=lat, longitude=lon))
+
+        # append the destination
+        global_path.waypoints.append(HelperLatLon(latitude=lat2, longitude=lon2))
+
+        if dst_file_path is not None:
+            # write to a new file
+            with open(dst_file_path, "w") as file:
+                writer = csv.writer(file)
+                writer.writerow(["latitude", "longitude"])
+                for waypoint in global_path.waypoints:
+                    writer.writerow([waypoint.latitude, waypoint.longitude])
```

Comment:
I see that you now write the global path to the file path. What do you see as the usecase for this? In my mind the node would only need to read from the files: the parameter could be used to quickly/temporarily tests a new path, and if we want to update or create a new file it makes sense to me for this to be a manual choice/process. Also with how it's currently implemented most of the time it will be writing the identical contents back into the file (for example when used in the plotting script)

Reply:
My thinking was that it might be nice to see the path, that was generated, in an easy to read and editable format. So I came up with the idea of writing the path back into the source file, as I don't see the use case for preserving that single destination point that was in there previously.

Reply:
That is a good point, but I also don't want to overwrite files. Maybe you could append the timestamp to the filename?

Reply:
Also, I needed a csv file that had a nice global path in it already, in order to plot it with the plotting script. So at some point we need to write a path to a csv file, in order to test out how the script plots a path from a csv.

Reply:
Also we probably don't want to write to the file when plotting, so you can set `dst_file_path` to `None`

Reply:
I don't think generate path will write to the csv over and over because it wont be called once the global path is more than one element, which should be the case after generate_path runs once.

But in that case, self.global path will get updated with the generated path once from generate_path() and then again from the csv file on the next callback run. So I will try to eliminate that redundancy.

Reply:
I like the filename + timestamp idea, I will add that